### PR TITLE
Improve tooltip and ticks strategy for parallel Chart

### PR DESCRIPTION
**Pull request by mgalloue**

### CODE REVIEW

File: src/models/scatter.js [hunk 1]
```diff
@@ -90,7 +91,7 @@ nv.models.scatter = function() {
             else
                 x.range(xRange || [0, availableWidth]);
 
-             if (chart.yScale().name === "o") {
```
Comment (mgalloue) [inline src/models/scatter.js]: Fix typo, name depend of how it's call on d3.js minified version

File: src/models/scatter.js [hunk 2]
```diff
@@ -174,8 +175,8 @@ nv.models.scatter = function() {
                                     var pX = getX(point,pointIndex);
                                     var pY = getY(point,pointIndex);
 
-                                    return [x(pX)+ Math.random() * 1e-4,
-                                            y(pY)+ Math.random() * 1e-4,
+                                    return [nv.utils.NaNtoZero(x(pX))+ Math.random() * 1e-4,
+                                            nv.utils.NaNtoZero(y(pY))+ Math.random() * 1e-4,
```
Comment (mgalloue) [inline src/models/scatter.js]: Allow log scale.

File: src/models/parallelCoordinates.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ nv.models.parallelCoordinates = function() {
         , height = null
         , x = d3.scale.ordinal()
         , y = {}
+        , undefinedValues = "undefined values"
```
Comment (mgalloue) [inline src/models/parallelCoordinates.js]: Allow undefined values customization

File: src/tooltip.js [hunk 4]
```diff
@@ -132,7 +132,7 @@ nv.models.tooltip = function() {
     var dataSeriesExists = function(d) {
         if (d && d.series) {
             if (d.series instanceof Array) {
-                return !!d.series.length;
+                return true;
```
Comment (mgalloue) [inline src/tooltip.js]: Allow tooltip even if array is empty

File: src/models/parallelCoordinates.js [hunk 5]
```diff
@@ -369,13 +371,34 @@ nv.models.parallelCoordinates = function() {
                         f.hasOnlyNaN = true;
                 });
                 dispatch.brushEnd(active, hasActiveBrush);
+            }           
+            function updateTicks() {
```
Comment (mgalloue) [inline src/models/parallelCoordinates.js]: Ticks visible are min and max for each axis, if there is a brush, brush extent is display.

File: src/models/parallelCoordinatesChart.js [hunk 6]
```diff
@@ -50,6 +50,20 @@ nv.models.parallelCoordinatesChart = function () {
             }
         };
 
+        tooltip.contentGenerator(function(data) {
```
Comment (mgalloue) [inline src/models/parallelCoordinatesChart.js]: Display a new tooltip for lines:
tooltipline
 And for dimensions:
tooltipdim

**Comment by mgalloue** [comment]
Hi @liquidpele, what can I do to help this PR get accepted?
Thanks a lot for your time.

File: src/models/parallelCoordinates.js (hunk 3 above)
Comment (liquidpele) [inline src/models/parallelCoordinates.js]: should probably be called "undefinedLabel" or something saying "Label" to match controlLables option which is used to change the name of control buttons.

File: src/tooltip.js (hunk 4 above)
Comment (liquidpele) [inline src/tooltip.js]: As long as it can't throw errors later due to it being an empty array...

File: src/models/parallelCoordinatesChart.js [hunk 7]
```diff
@@ -222,11 +233,20 @@ nv.models.parallelCoordinatesChart = function () {
...
-                color: evt.color
-            };
-            tooltip.data(evt).hidden(false);
+                color: evt.color,
+                series: []
+             }      
+            if(evt.values){
+                Object.keys(evt.values).forEach(function (d) {
+                    var dim = evt.dimensions.filter(function (dd) {return dd.key === d;})[0];
+                    if(dim)
```
Comment (liquidpele) [inline src/models/parallelCoordinatesChart.js]: Please add brackets for all if statements, even if only a one liner.

**Comment by liquidpele** [comment]
Looks like you have merge conflicts you need to resolve as well.   Sorry for the delay, I wasn't able to look at nvd3 for a few months there.

**Comment by mgalloue** [comment]
@liquidpele, I close this PR, the updated one is ready to merge on PR [#1459](https://github.com/novus/nvd3/pull/1459)

**Pull request closed without merging**